Client/Frontend/Browser/RecentlyClosedTabs.swift

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+public protocol RecentlyClosedTabs {
+    func addTab(url: NSURL?, title: String?, faviconURL: String?)
+    func getData() -> [String:[String]]?
+    func clearData()
+    func count() -> Int
+}
+
+public class RecentlyClosedTabsStore: RecentlyClosedTabs {
+    let prefs: Prefs
+
+    public init(prefs: Prefs) {
+        self.prefs = prefs
+    }
+
+    public func addTab(url: NSURL?, title: String?, faviconURL: String?) {
```

Comment:
`url` shouldn't be optional. No point in storing a closed tab if we don't even have a URL to open!